orbs/pulumi.yml

```
@@ -42,12 +42,36 @@ commands:
       - run: 
           name: "Install Pulumi CLI, if needed"
           command: |
-              if [ << parameters.version >> == "latest" ]; then
+              PARAM_VERSION="<< parameters.version >>"
+
+              if command -v pulumi > /dev/null; then
+                # Strip 'v' prefix if present (e.g., v3.101.0 -> 3.101.0)
+                INSTALLED_VERSION=$(pulumi version | tr -d 'v')
+                echo "Found installed Pulumi version: $INSTALLED_VERSION"
+
+                # Determine the target version for comparison
+                if [ "$PARAM_VERSION" == "latest" ]; then
+                  TARGET_VERSION=$(curl -s https://www.pulumi.com/latest-version)
+                  echo "Latest available version is: $TARGET_VERSION"
+                else
+                  TARGET_VERSION="$PARAM_VERSION"
+                fi
+
+                # Compare and exit if they match
+                if [ "$INSTALLED_VERSION" == "$TARGET_VERSION" ]; then
+                  echo "Pulumi $INSTALLED_VERSION is already installed and matches requested version. Skipping install."
+                  exit 0
+                else
+                  echo "Version mismatch (Installed: $INSTALLED_VERSION, Requested: $TARGET_VERSION). Proceeding with installation."
+                fi
+              fi
+
+              if [ "$PARAM_VERSION" == "latest" ]; then
```

```suggestion
              if [ "$PARAM_VERSION" == "latest" ] || [ -z $PARAM_VERSION ]; then
```
If no version is given, but we don't have the CLI installed I think we want to continue installing the latest version? I don't think the `--version` parameter is happy with no argument (though this seems to be a pre-existing problem, so feel free to ignore.